### DISCUSSION TITLE
how to send http request synchronously using supabase js library?

### DISCUSSION THREAD

**Discussion by saman1324** [supabase/supabase#6043] [Questions]
hello.
im creating a chat app. i want to send users messages synchronously. why? because for example user sends 4 messages. these 4 messages will be sent asynchronously. so it may cause problems. for example message number 4 will be sent to server faster and supabase will store that first. then, message number 2 will be arrived and then message 1 and 3. so the user who is seeing those messages, sees message number 4, 2, 1, 3 respectively and its not good at all! because of it i want to send users messages synchronously. so in this case, message number 1 will be sent to server. if it arrived to server or it failed, message number 2 will be sent. same for message number 2 and then message number 3 will be sent and it is the same for all sending messages. so user who is seeing the senders messages sees message 1, 2, 3, 4 respectively. i want it!
so do you have any idea to send users messages synchronously?
thank you very much for your help.

**Comment by vinliao** [comment]
Timestamp the message on the chat client, then sort based on that timestamp (don't use the Postgres' timestamp).
